rasa/utils/tensorflow/model_data.py

```diff
@@ -458,21 +514,37 @@ def _get_shapes_types(self) -> Tuple:
         types = []
         shapes = []
 
-        def append_shape(features: np.ndarray) -> None:
-            if isinstance(features[0], scipy.sparse.spmatrix):
-                # scipy matrix is converted into indices, data, shape
-                shapes.append((None, features[0].ndim + 1))
-                shapes.append((None,))
-                shapes.append((features[0].ndim + 1))
-            elif features[0].ndim == 0:
-                shapes.append((None,))
-            elif features[0].ndim == 1:
-                shapes.append((None, features[0].shape[-1]))
+        def append_shape(_features: Union[Features_4D, Features_3D]) -> None:
```

Comment:
it looks like in the rest of the code, `features` correspond to `Union[List[Features_4D], List[Features_3D]]` shall we create type for it?
and rename `_features` here to smth more explicit?